ext/standard/streamsfuncs.c

```diff
@@ -616,97 +616,150 @@ PHP_FUNCTION(stream_get_wrappers)
 }
 /* }}} */
 
-/* {{{ stream_select related functions */
-static int stream_array_to_fd_set(zval *stream_array, fd_set *fds, php_socket_t *max_fd)
-{
-	zval *elem;
-	php_stream *stream;
-	int cnt = 0;
+/* {{{ fd_bigset related macros and typedefs to remove FD_SETSIZE limitation in stream_select */
+typedef struct {
+#ifdef __USE_XOPEN
+    char *fds_bits;
+#else
+    char *__fds_bits;
+#endif
+    size_t size;
+} fd_bigset;
+
+#ifdef __USE_XOPEN
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) ((set)->fds_bits[(fd) / 8] |= (1 << ((fd) % 8)))
+#define FD_BIGSET_ISSET(fd, set) ((set)->fds_bits[(fd) / 8] & (1 << ((fd) % 8)))
+#define FD_BIGSET_CLR(fd, set) ((set)->fds_bits[(fd) / 8] &= ~(1 << ((fd) % 8)))
+#define FD_BIGSET_FREE(set) do { \
+    if ((set)->fds_bits) { \
+        efree((set)->fds_bits); \
+        (set)->fds_bits = NULL; \
+    } \
+    (set)->size = 0; \
+} while (0)
+#else
+#define FD_BIGSET_ZERO(set, num_fds) do { \
+    (set)->size = (num_fds + 7) / 8; \
+    (set)->__fds_bits = (char *)ecalloc((set)->size, sizeof(char)); \
+} while (0)
+#define FD_BIGSET_SET(fd, set) ((set)->__fds_bits[(fd) / 8] |= (1 << ((fd) % 8)))
+#define FD_BIGSET_ISSET(fd, set) ((set)->__fds_bits[(fd) / 8] & (1 << ((fd) % 8)))
+#define FD_BIGSET_CLR(fd, set) ((set)->__fds_bits[(fd) / 8] &= ~(1 << ((fd) % 8)))
+#define FD_BIGSET_FREE(set) do { \
+    if ((set)->__fds_bits) { \
+        efree((set)->__fds_bits); \
+        (set)->__fds_bits = NULL; \
+    } \
+    (set)->size = 0; \
+} while (0)
+#endif
 
-	if (Z_TYPE_P(stream_array) != IS_ARRAY) {
-		return 0;
-	}
+int stream_array_to_fd_bigset(zval *stream_array, fd_bigset *set, php_socket_t *max_fd) {
+    zval *elem;
+    php_stream *stream;
+    int cnt = 0;
 
-	ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(stream_array), elem) {
-		/* Temporary int fd is needed for the STREAM data type on windows, passing this_fd directly to php_stream_cast()
-			would eventually bring a wrong result on x64. php_stream_cast() casts to int internally, and this will leave
-			the higher bits of a SOCKET variable uninitialized on systems with little endian. */
```

Comment:
Are comments like these no longer relevant? They seem to share some useful information here.

Reply:
That comment is copied many places in the source code, but on Windows the php_select function is a macro to another implementation afaik.

Reply:
I have put the comment back in.